packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -277,6 +277,55 @@ correct log retention period (never expire, by default).
 *Further note* that, if the log group already exists and the `logRetention` is not set, the custom resource will reset
 the log retention to never expire even if it was configured with a different value.
 
+### FileSystem Access
+
+You can configure a function to mount an Amazon Elastic File System (Amazon EFS) to a
+directory in your runtime environment with the `filesystems` property. To access Amaozn EFS
+from lambda function, the Amazon EFS access point will be required. 
+
+Amazon EFS _access points_ are application-specific entry points into an EFS file system that
+make it easier to manage application access to shared datasets. Access points can enforce a user
+identity, including the user's POSIX groups, for all file system requests that are made through
+the access point. Access points can also enforce a different root directory for the file system
+so that clients can only access data in the specified directory or its subdirectories. See
+[Working with Amazon EFS Access Points](https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html) for more details.
+
+The following sample enables the lambda function to mount the Amazon EFS filesystem via the `accessPoint` to `/mnt/msg` in the runtime environment. We have to create an access point in Amaozn EFS and make sure Lambda function has
+read write access to it.
+
+```ts
+// create a new Amaozn EFS filesystem
+const fileSystem = new efs.FileSystem(stack, 'Efs', { vpc });
+
+// create a new access point from the filesystem
+const accessPoint = fileSystem.addAccessPoint('AccessPoint', {
+  // set /mnt/msg as the root of the access point
+  path: '/mnt/msg',
+  // as /mnt/msg does not exist, the efs will create the directory with the following createAcl
+  createAcl: {
```

Comment:
Asking again, can we omit this? Are these the defaults?

Reply:
Hi @eladb 

After some investigation, yes we can create the accessPoint simply with

```ts
// create an access point and expose the root of the filesystem
const accessPoint = fileSystem.addAccessPoint('AccessPoint');
```

However, it indicates that the AccessPoint will expose the root directory of the efs filesystem and lambda will not be able to access it with no POSIX user enforcement and will simply throws permission denied.

So now I am proposing the following sample

```ts
// create an access point and expose the root of the filesystem
const accessPoint = fileSystem.addAccessPoint('AccessPoint', {
  createAcl: {
    ownerGid: '1001',
    ownerUid: '1001',
    permissions: '750',
  },
  path: '/export/lambda',
  posixUser: {
    gid: '1001',
    uid: '1001',
  }
});
```
And in Lambda:

```ts
  filesystems: [
    // mount the access point to /mnt/msg in the lambda runtime enironment
    lambda.FileSystem.fromEfsAccessPoint(accessPoint, '/mnt/msg'),
  ],
```

In this case:

1. A new `AccessPoint` will be created exposing the `/export/lambda` from the efs filesystem to the client(lambda function in this case).

2. When `/export/lambda` was created, it will be created with the permission defined in `createAcl`, which is uid=1001 and gid=1001 with permission `750`.  It's not default or any recommended value but just a common practice and was used in our [blog post](https://aws.amazon.com/tw/blogs/aws/new-a-shared-file-system-for-your-lambda-functions/).

3. Now the `/export/lambda` was created and exposed and `posixUser` was defined, the lambda function will be able to mount this access point with the posixUser identity enforcement, which is also uid=1001 and gid=1001 and eventually be able to mount the access point to`/mnt/msg` in its runtime environment.


Looks like this is mandatory as a working sample and I didn't see anything we can concise or skip any further. WDYT?

Reply:
What happens if you only specify the `path`:

```ts
const accessPoint = fileSystem.addAccessPoint('AccessPoint', {
  path: '/export/lambda'
});
```

Reply:
If only path specified without CreateAcl then there will be two conditions:

1. The path already exists then everything is fine but we still have to specify posixUser otherwise the Lambda might not be able to access it(depends on the directory permission of the path)

2. The path actually doesn't exist then createAcl will be required so efs will create it with the settings in createAcl for us. If createAcl not defined, cloudformation will throw error.

As we can't check if path already exists from CDK, we have a note about this in the AccessPoint doc string.

Reply:
Okay, let's leave as is for now. Feels like we can do better but I am failing to understand and it's probably okay for this example.